src/native/libs/System.Native/pal_io.c

```diff
@@ -1189,19 +1189,22 @@ static bool SupportsCopyFileRange()
     {
         isSupported = -1;
 
+#ifdef TARGET_LINUX
         // Avoid known issues with copy_file_range that are fixed in Linux 5.3+ (https://lwn.net/Articles/789527/).
         struct utsname name;
         if (uname(&name) == 0)
         {
-            if (name.release[1] != '.' ||
-                name.release[0] > '5'  ||
-                (name.release[0] == '5' &&
-                    (name.release[3] != '.' ||
-                     name.release[2] >= '3')))
+            if (name.release[0] && name.release[1] &&
+                (name.release[1] != '.' ||
+                 name.release[0] > '5'  ||
+                 (name.release[0] == '5' && name.release[2] && name.release[3] &&
+                     (name.release[3] != '.' ||
+                      name.release[2] >= '3'))))
```

Comment:
I can already see how this breaks when Linux 10 gets released 😅

Reply:
Perhaps something like https://godbolt.org/z/odYxG7sT5 would be cleaner / resilient?

Reply:
In theory it needs some improvement for short release strings causing false negatives. But such short strings are not expected.

10.y meets the length requirement.

Reply:
Ah, I found the sscanf pattern a bit more idiomatic and has precedence; used in other places like:

```
src/coreclr/dlls/dbgshim/dbgshim.cpp
src/mono/mono/utils/mono-mmap.c
src/native/libs/System.Globalization.Native/pal_icushim.c
```

Reply:
Since it's a question of style, I'm going to pass it to @adamsitnik and @jkotas. If they prefer the `sscanf` I'll rewrite it.

Reply:
It's not just style. The current check will literally say that `copy_file_range` is not supported on Linux 10...

Reply:
ok, I'll change it to `sscanf` in case Linux stops using minor and patch version numbers.

Reply:
It's not about the minor/patch numbers. `10.0.0` just happens to use double digits for the first number and thus fails the `> '5'` check.

Reply:
>  I'm going to pass it to @adamsitnik and @jkotas.

I think `sscanf` would be much easier to read.

Reply:
> `name.release[1] != '.' ||`

Ah, sorry I missed that. The line was conveniently hidden just outside of the view on the GitHub conversations page.